Assignment/palindrome_test.c

```diff
@@ -20,102 +20,101 @@
 }
 
-void PalindromeTest () {
-   char* input1[] = { "vini", "I did,did I", "Ma*)d(*am", "Was it a car or a cat I saw?", "Hello", "Don't nod" };
-   char* expectedOutput1[] = { "Not a Palindrome", "Palindrome", "Palindrome", "Palindrome", "Not a Palindrome", "Palindrome" };
-   int testCount = sizeof (input1) / sizeof (input1[0]);
-   printf ("------Palindrome Check------\n"
+void TestCases () {
+   char* inputs[] = { "vini", "I did,did I", "Ma*)d(*am", "@@@##!!!", "Mam!",
+                      "Was it a car or a cat I saw?", "Don't nod", "Hello",
+                      "4334", "654", "34543", "-121", "*-23" ,"?@343", "6", " " };
+   char* expOutputs[] = { "Not a Palindrome", "Palindrome", "Palindrome", "Invalid Input",
+                          "Palindrome", "Palindrome", "Palindrome", "Not a Palindrome",
+                          "Palindrome", "Not a Palindrome", "Palindrome", "Not a Palindrome",
+                          "Invalid Input", "Invalid Input", "Palindrome", "Invalid Input" };
+   int numOfTestCases = sizeof (inputs) / sizeof (inputs[0]);
+   printf ("\n-------Testcases for String and Integer Input------\n\n"
            "| %-30s | %-20s | %-20s | %-10s |\n"
-           "|--------------------------------|----------------------|----------------------|------------------|\n",
+           "|--------------------------------|----------------------|"
+           "----------------------|------------------|\n",
            "Input", "Expected Output", "Actual Output", "Test Case Result");
-   int i = 0, actualOutput;
-   for (i = 0; i < testCount; i++) {
-      actualOutput = PalindromeCheck (input1[i]);
-      PrintRes (input1[i], expectedOutput1[i], actualOutput);
-   }
-   printf ("\n\n");
-}
-
-void ReverseNumPalindromeTest () {
-   char* input2[] = { "123", "4334", "654", "34543", "-121","*-23" ,"?@343", "6", " " };
-   char* expectedOutput2[] = { "Not a Palindrome", "Palindrome", "Not a Palindrome", "Palindrome", "Not a Palindrome","Invalid Input","Invalid Input", "Palindrome", "Invalid Input" };
-   int testCount = sizeof (input2) / sizeof (input2[0]);
-   printf ("------ReverseNumber Palindrome Check------\n"
-           "| %-30s | %-20s | %-20s | %-10s |\n"
-           "|--------------------------------|----------------------|----------------------|------------------|\n",
-           "Input", "Expected Output", "Actual Output", "Test Case Result");
-   int i = 0, num, reversed = 0, actualOutput;
-   for (i = 0; i < testCount; i++) {
-      if (!IsValid (input2[i])) {
-         const char* actualRes = "Invalid Input";
-         const char* res = (strcmp (actualRes, expectedOutput2[i]) == 0) ? GREEN_TEXT "Pass" RESET_TEXT : RED_TEXT "Fail" RESET_TEXT;
-         printf ("| %-30s | %-20s | %-20s | %-27s |\n", input2[i], expectedOutput2[i], actualRes, res);
-         continue;
+   for (int i = 0; i < numOfTestCases; i++) {
+      int actualOutput;
+      const char* actualRes;
+      if (IsValid (inputs[i])) {

```

Comment:
Can use ternary operator to simplify this if-else statement. Try to simplify using ternary operator wherever possible.